.ai-doc/50-migration-angular/migration-patterns.md

```diff
@@ -0,0 +1,1355 @@
+resolve: {
+  goBack: /* @ngInject */ (
+    $state,
+    serviceName,
+    alertSuccess,
+    alertError,
+  ) => ({ success, error, stateName, reload } = {}) => {
+    const name = stateName || '^';
+    const prms = { serviceName };
+    const opts = {
+      reload: reload === true || (Boolean(success) && reload !== false),
+    };
+    return $state.go(name, prms, opts).then((result) => {
+      if (success) {
+        alertSuccess(success);
+      }
+      if (error) {
+        alertError(error);
+      }
+      return result;
+    });
+  },
+}
+```
+
+**React:**
+```typescript
+// In component
+const handleGoBack = (options?: {
+  success?: string;
+  error?: Error;
+  reload?: boolean;
+}) => {
+  if (options?.success) {
+    // Show success notification
+    showNotification({ type: 'success', message: options.success });
+  }
+  if (options?.error) {
+    // Show error notification
+    showNotification({ type: 'error', message: options.error.message });
+  }
+  
+  if (options?.reload) {
+    // Invalidate queries to reload data
+    queryClient.invalidateQueries({ queryKey: ['nasha-detail', serviceName] });
+  }
+  
+  navigate(`/${serviceName}`, { replace: true });
+};
+```
+
+**Mapping Strategy:**
+- Navigation functions become component handlers
+- Alert system becomes notification system
+- Reload logic becomes cache invalidation
+- Promise-based becomes callback-based
+
+### Edge Case 4: managerListLayout Mapping
+
+**AngularJS (directory.routing.js):**
+```javascript
+component: 'managerListLayout',
+resolve: {
+  columnConfig: /* @ngInject */ ($translate) => ({
+    data: [
+      {
+        label: $translate.instant('key'),
+        property: 'field',
+        serviceLink: true,
+        hidden: false,
+        format: (row) => formatValue(row.field),
+      }
+    ],
+  }),
+  topbarOptions: /* @ngInject */ ($translate, $state, atInternet) => ({
+    cta: {
+      type: 'button',
+      label: $translate.instant('order_label'),
+      onClick: () => $state.go('nasha.order'),
+    },
+  }),
+}
+```
+
+**React (Listing.page.tsx):**
+```typescript
+<BaseLayout
+  header={{
+    title: t('listing_title'),
+    changelogButton: <ChangelogMenu />,
+    guideMenu: <GuideMenu />,
+  }}
+>
+  <Button onClick={handleOrderClick}>
+    {t('order_label')}
+  </Button>
+  <Datagrid
+    columns={columns}
+    data={data}
+    // ... other props
+  />
+</BaseLayout>
+```
+
+**Mapping Strategy:**
+1. `component: 'managerListLayout'` → `<BaseLayout>` + `<Datagrid>`
+2. `columnConfig.data` → `columns` array with `DatagridColumn` type
+3. `topbarOptions.cta` → `<Button>` in header area
+4. `serviceLink: true` → Custom `cell` renderer with `<Link>`
+5. `format` function → Custom `cell` renderer
+6. `ListLayoutHelper.stateResolves` → Custom hooks for pagination, sorting, filtering
+
+## 🔗 AAPI Integration Pattern
+
+### When to Use AAPI
+
+Use AAPI when AngularJS code uses:
+- `OvhApi*.Aapi()`
+- `.resetCache()` (though not needed in React)
```

Comment:
It's needed to avoid fetching wrong data. Appropriate header must be sent to discard cache and fetch fresh data from backend.